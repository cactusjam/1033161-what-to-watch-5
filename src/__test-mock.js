import {NameSpace} from "./store/reducers/root-reducer.js";

export const TEST_MOCK_MOVIE = {
  background: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
  backgroundColor: `#73B39A`,
  cover: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
  description: `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
  director: `Nicolas Winding Refn`,
  duration: 92,
  genre: `Action`,
  id: 1,
  isFavorite: true,
  poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
  promo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  rating: {
    score: 3.6,
    countOfVotes: 109661
  },
  releaseYear: 2008,
  starring: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
  title: `Bronson`,
  video: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`
};

export const TEST_MOCK_COMMENT = {
  author: `Christina`,
  date: `2020-10-09T13:38:44.769Z`,
  id: 1,
  rating: 3.3,
  reviewText: `I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quali.`
};

export const TEST_MOCK_USER = {
  id: 1,
  email: `qqq@gmail.com`,
  userAvatar: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/9.jpg`,
  authorizationStatus: `AUTH`,
  password: `666`,
  errorEmail: ``,
  errorPassword: ``
};

export const TEST_MOCKS = {
  id: 1,
  noop: () => {},
  movie: TEST_MOCK_MOVIE,
  movies: [TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE],
  genresFilter: [`All genres`, `Action`, `Adventure`, `Comedy`, `Crime`, `Drama`, `Fantasy`, `Thriller`],
  genreFilter: `All genres`,
  review: TEST_MOCK_COMMENT,
  userData: {
    id: 1,
    email: `qqq@gmail.com`,
    userAvatar: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/9.jpg`,
    authorizationStatus: `AUTH`
  },
  unauthorizedUserData: {
    id: 1,
    email: ``,
    userAvatar: ``,
    authorizationStatus: `NO_AUTH`
  },
  reviews: [TEST_MOCK_COMMENT, TEST_MOCK_COMMENT, TEST_MOCK_COMMENT],
};

export const TEST_MOCK_STORE = {
  [NameSpace.USER]: {
    id: 1,
    email: `qqq@gmail.com`,
    authorizationStatus: `AUTH`,
    userAvatar: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/9.jpg`,
  },
  [NameSpace.GENRES]: {
    activeFilterGenre: `Adventure`
  },
  [NameSpace.DATA]: {
    movies: [TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE],
    reviews: [TEST_MOCK_COMMENT, TEST_MOCK_COMMENT, TEST_MOCK_COMMENT],
    currentMovie: TEST_MOCK_MOVIE,
    promoMovie: TEST_MOCK_MOVIE,
    favorites: [TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE],
    defaultFilmsCount: 8,
    isDataSending: false,
    isDataSendError: false
  },
};
