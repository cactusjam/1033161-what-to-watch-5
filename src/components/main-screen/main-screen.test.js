import React from "react";
import renderer from "react-test-renderer";
import MainScreen from "./main-screen";
import {TEST_MOCK_STORE, TEST_MOCKS} from "../../__test-mock.js";
import {MemoryRouter} from "react-router-dom";
import configureMockStore from "redux-mock-store";
import {Provider} from "react-redux";

const mockStore = configureMockStore();
const store = mockStore(TEST_MOCK_STORE);

describe(`MainScreen`, () => {
  it(`Should PageMain render correctly`, () => {
    const tree = renderer
    .create(
        <Provider store={store}>
          <MemoryRouter>
            <MainScreen
              defaultFilmsCount={8}
              filteredMovies={TEST_MOCKS.movies}
              genreFilter={TEST_MOCKS.genreFilter}
              movies={TEST_MOCKS.movies}
              onPlayButtonClick={TEST_MOCKS.noop}
              onGenreFilterChange={TEST_MOCKS.noop}
              promoMovie={TEST_MOCKS.movie}
            />,
          </MemoryRouter>
        </Provider>
        , {
          createNodeMock: () => {
            return {};
          }
        }
    )
  .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
