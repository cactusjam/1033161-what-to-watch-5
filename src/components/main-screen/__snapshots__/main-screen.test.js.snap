// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainScreen Should PageMain render correctly 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Bronson"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg"
      />
    </div>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          href="/mylist"
          onClick={[Function]}
        >
          <div
            className="user-block__avatar"
          >
            <img
              alt="User avatar"
              height="63"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/9.jpg"
              width="63"
            />
          </div>
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="Bronson poster"
            height="327"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Bronson
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Action
            </span>
            <span
              className="movie-card__year"
            >
              2008
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player/1"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#in-list"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item "
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item "
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
          >
            Action
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      />
      
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
  ",",
]
`;
